pontoon/base/models.py

```diff
@@ -3282,6 +3282,28 @@ def for_checks(self, only_db_formats=True):
 
         return translations
 
+    def bulk_mark_changed(self):
+        changed_entities = {}
+        existing = ChangedEntityLocale.objects.values_list(
+            "entity", "locale"
+        ).distinct()
+
+        for translation in self:
```

Comment:
You can `self.exclude()` DB-projects here instead of using the if-block below. That will also save us the need to prefetch data.